// shared for all messages
message ModemCommon
{
  optional uint32 src = 1;
  optional uint32 dest = 2;
  optional string iso_time = 3;
  optional double unix_time = 4;
  optional uint32 rate = 5;
  optional string raw = 6;
}

// data 
message ModemDataTransmission
{
  optional ModemCommon header = 1;
  optional bool ack_request = 2;
  optional uint32 max_bytes = 3;
  optional uint32 frame = 4;
  optional string data = 5;
}

message ModemDataAck
{
  optional ModemCommon header = 1;
  optional ModemDataTransmission orig_msg = 2;
}

message ModemDataRequest
{
  optional ModemCommon header = 1;

  optional bool ack_requested = 2;
  optional uint32 max_bytes = 3;
  optional uint32 frame = 4;
}

// ranging
enum RangingType
{
  MODEM = 1;
  REMUS_LBL = 2;
}

message ModemRangingReply
{
  optional ModemCommon header = 1;
  // seconds
  repeated double one_way_travel_time = 2;  
  optional RangingType type = 3;
}

message ModemRangingRequest
{
  optional ModemCommon header = 1;
  optional RangingType type = 2;
}
